allennlp/data/dataset_readers/coreference_resolution/conll.py

```diff
@@ -0,0 +1,197 @@
+import logging
+import re
+import collections
+from typing import Any, Dict, List, Optional, Tuple
+
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset import Dataset
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import Field, ListField, TextField, IndexField, MetadataField, SequenceLabelField
+from allennlp.data.instance import Instance
+from allennlp.data.tokenizers import Token
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+@DatasetReader.register("coref")
+class ConllCorefReader(DatasetReader):
+    def __init__(self,
+                 max_span_width: int,
+                 token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        self._max_span_width = max_span_width
+        self._token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+        self._begin_doc_regex = re.compile(r"#begin document \((.*)\); part (\d+)")
+
+    @overrides
+    def read(self, file_path: str):
+        # if `file_path` is a URL, redirect to the cache
+        file_path = cached_path(file_path)
+
+        logger.info("Reading file at %s", file_path)
+        instances = []
+        with open(file_path) as dataset_file:
```

Comment:
Are we missing an outer loop over documents in the dataset here? Seems like this just handles a single file.

Reply:
Oh, maybe your bash script compiles it into a single file. My bad.

Reply:
Added docstring.